tools/tests/py2fgen/test_dycore_wrapper.py

```diff
@@ -0,0 +1,1593 @@
+            ndyn_substeps=ndyn_substeps,
+            idyn_timestep=jstep_init_fortran,
+            nnow=nnow,
+            nnew=nnew,
+        )
+
+        # Check input arguments to SolveNonhydro.time_step
+        captured_args, captured_kwargs = mock_init.call_args
+
+        result, error_message = compare_objects(
+            captured_kwargs["diagnostic_state_nh"], expected_diagnostic_state_nh
+        )
+        assert result, f"Diagnostic State comparison failed: {error_message}"
+
+        result, error_message = compare_objects(
+            captured_kwargs["prognostic_state_ls"], expected_prognostic_state_ls
+        )
+        assert result, f"Prognostic State comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["prep_adv"], expected_prep_adv)
+        assert result, f"Prep Advection comparison failed: {error_message}"
+
+        result, error_message = compare_objects(
+            captured_kwargs["divdamp_fac_o2"], expected_initial_divdamp_fac
+        )
+        assert result, f"Divdamp Factor comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["dtime"], expected_dtime)
+        assert result, f"dtime comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["l_recompute"], expected_recompute)
+        assert result, f"Recompute flag comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["l_init"], expected_linit)
+        assert result, f"Init flag comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["lclean_mflx"], expected_clean_mflx)
+        assert result, f"Clean MFLX flag comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["lprep_adv"], expected_lprep_adv)
+        assert result, f"Prep Advection flag comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["nnew"], expected_nnew)
+        assert result, f"nnew comparison failed: {error_message}"
+
+        result, error_message = compare_objects(captured_kwargs["nnow"], expected_nnow)
+        assert result, f"nnow comparison failed: {error_message}"
+
+        result, error_message = compare_objects(
+            captured_kwargs["at_first_substep"], expected_at_first_substep
+        )
+        assert result, f"First Substep comparison failed: {error_message}"
+
+        result, error_message = compare_objects(
+            captured_kwargs["at_last_substep"], expected_at_last_substep
+        )
+        assert result, f"Last Substep comparison failed: {error_message}"
+
+
+@pytest.mark.datatest
+@pytest.mark.parametrize("istep_init,jstep_init, istep_exit,jstep_exit", [(1, 0, 2, 0)])
+@pytest.mark.parametrize(
+    "experiment,step_date_init, step_date_exit",
+    [
+        (
+            dt_utils.REGIONAL_EXPERIMENT,
```

Comment:
Did you ever try to run the `GLOBAL_EXPERIMENT`?

Reply:
No, because for the global experiment I have to handle certain fields in a special way just like you do in the serializer with `@optionally_registered` since some fields used in the local experiment are not used in the global experiment (or have to be initialized with zeroes. We can look at that as a next step.